silex/lang.lua

```diff
@@ -10,117 +10,117 @@ SILE.X = SILE.X or {}
 ---@param langbcp47 string                a valid BCP47 canonical language
```

Comment:
Let's get this BCP-47 handling upstreamed ASP. I think with the language module overhaul I was just working on this should be pretty easy to add in on top of that.

Reply:
For the record, the stuff here is just a "minimal" subset of https://github.com/sile-typesetter/sile/pull/1641 (so raised up-streamed end 2022), so I could use BCP47 specifiers in **markdown.sile**.
(And the `SILE.X` stuff above was just for **smartquotes.sile** to be happy with some API compatibility level on these language hacks,  for what it does - e.g. it uses `SILE.X.forLanguage` (IIRC) - but not necessarily linked to the version of **silex** itself.

Reply:
Great, if what is here is a subset of that PR, I'll just work from the PR and not try to review the implementation here.